### PR TITLE
Add support for cloudfoundry CUPS.

### LINKED ISSUE #306: Setup User Provided Services (UPS) for all CALC deployments on cloud.gov
To persist settings  (such as `SECRET_KEY`) across deploys, we should use a User Provided Service to store settings/credentials.

Documentation: https://docs.cloudfoundry.org/devguide/services/user-provided.html

Also see https://github.com/18f/team-api-server#on-cloudgov

### PR DESCRIPTION
**Pull request by toolness**
This fixes #306.

The `load_cups_from_vcap_services()` function was largely inspired by [python-dotenv](https://github.com/theskumar/python-dotenv#readme).

The documentation is largely cribbed from [team-api-server](https://github.com/18F/team-api-server/blob/master/README.md#on-cloudgov)'s.

### CODE REVIEW

File: deploy.md [hunk 1]
```diff
@@ -23,6 +23,41 @@ To start, target the org and space you want to work with. For example, if you wa
...
 
+#### Environment Variables
+
+For cloud.gov deployments, this project makes use of a
+[Custom User Provided Service (CUPS)][CUPS] to get its configuration
+variables, instead of using the local environment. You will need to create a
+CUPS called `calc-env`, provide 'credentials' to it, and link it to the
+application instance.
+
+First, create a file called `credentials.json` with all the configuration
```
Comment (jseppi) [inline deploy.md]: Should probably add `credentials.json` to `.gitignore`

**Comment by jseppi** [comment]
I noticed some very minor pep8 issues in the new files (trailing spaces in tests.py and only one blank line before the `def` in settings_utils.py).

Other than that 👍

**Comment by toolness** [comment]
Good catches! Just fixed them and also added `flake8` to `requirements.txt` so we can do linting as part of the test suite in the future, and so it's in the virtualenv/docker container of devs automatically.

**Comment by jseppi** [comment]
👍

**Pull request merged**